### PR TITLE
⭐️ unit-testing for nodes and workload configuration

### LINKED ISSUE #162: Write test cases for node scanning functions
Similar to how the webhook setup functions have test cases in https://github.com/mondoohq/mondoo-operator/blob/main/controllers/webhooks_test.go , follow that pattern to write up test cases for the code in https://github.com/mondoohq/mondoo-operator/blob/main/controllers/nodes.go

### LINKED ISSUE #163: Write test cases for workload scanning functions
Similar to how the webhook setup functions have test cases in https://github.com/mondoohq/mondoo-operator/blob/main/controllers/webhooks_test.go , follow that pattern to write up test cases for the code in https://github.com/mondoohq/mondoo-operator/blob/main/controllers/workloads.go

### PR DESCRIPTION
**Pull request by jensgrnb**
Signed-off-by: jensgrnb <jens.grunborg@endocode.com>

closes #162, closes #163

### CODE REVIEW

File: controllers/nodes_test.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+var _ = Describe("nodes", func() {
+	const (
+		name      = "nodes"
+		namespace = "default"
+		timeout   = time.Second * 10
+		duration  = time.Second * 10
+		interval  = time.Millisecond * 250
+	)
+	Context("When deploying the operator with nodes enabled", func() {
```
Comment (joelddiaz) [inline controllers/nodes_test.go:44]: Would be nice to see a test where when there is an existing DaemonSet, and the MondooAuditConfig has node scanning disabled, that the DaemonSet is deleted.

And a test exercising the ability to override the image/tags.

File: controllers/workloads_test.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+var _ = Describe("workloads", func() {
+	const (
+		name      = "workloads"
+		namespace = "default"
+		timeout   = time.Second * 10
+		duration  = time.Second * 10
+		interval  = time.Millisecond * 250
+	)
+	Context("When deploying the operator with workloads enabled", func() {
```
Comment (joelddiaz) [inline controllers/workloads_test.go:44]: Similar comment here about ensuring that nothing unexpected is running when Workload.Enabled == false, plus some image/tag test validation.

**Review by joelddiaz — COMMENTED** [review]
The approach is definitely different than what I'm used to, but it looks sensible to me. How do you go about walking through these tests in a debugger?

File: controllers/nodes_test.go (hunk 1 above)
Comment (jensgrnb) [reply controllers/nodes_test.go:44]: Yep sounds reasonable, I'm still in the discovery phase with ginkgo

**Comment by joelddiaz** [comment]
:+1: LGTM . These first test cases look good enough to merge as-is.

**Pull request merged**